syntax = "proto3";

package tensorflow.embedding;

enum StorageType {
  // none
  DEFAULT = 0;

  // one level
  DRAM = 1;
  PMEM_MEMKIND = 2;
  PMEM_LIBPMEM = 3;
  SSDHASH = 4;
  LEVELDB = 5;
  HBM = 6;

  // two level
  DRAM_PMEM = 11;
  DRAM_SSDHASH = 12;
  HBM_DRAM = 13;
  DRAM_LEVELDB = 14;

  // three level
  DRAM_PMEM_SSDHASH = 101;
  HBM_DRAM_SSDHASH = 102;

}

enum CopyBackFlag {
  NOT_COPYBACK = 0;
  COPYBACK = 1;
  COPYBACK_AND_DESTROY = 2;
}

enum SlotType {
  EMBEDDING_VARIABLE = 0;
  VARIABLE = 1;
}

enum CacheStrategy {
  LRU = 0;
  B4LRU = 1;
  B8LRU = 2;
  B16LRU = 3;
  B32LRU = 4;
  B48LRU = 5;
  B64LRU = 6;
  LFU = 7;
  B4LFU = 8;
  B8LFU = 9;
  B16LFU = 10;
  B32LFU = 11;
  B48LFU = 12;
  B64LFU = 13;
  ShardedLRU = 14;
}

enum EmbeddingVariableType {
  IMMUTABLE = 0;
  MUTABLE = 1;
}

enum ValuePtrStatus {
  OK = 0;
  IS_DELETED = 1;
  NOT_IN_DRAM = 2;
}

enum IsSetInitialized {
  NOT_SET_INITAILIZED = 0;
}
